peak/mapper/formula_constructor.py

```diff
@@ -0,0 +1,67 @@
+import operator
+from functools import partial, reduce, lru_cache
+
+or_reduce = partial(reduce, operator.or_)
+and_reduce = partial(reduce, operator.and_)
+
+
+class FormulaConstructor:
+    indent = "|   "
+
+    @staticmethod
+    def _to_smt(v):
+        if isinstance(v, FormulaConstructor):
+            return v.to_smt()
+        return v
+
+    @staticmethod
+    def print_value(v, ts):
```

Comment:
would prefer this returns strings instead of printing directly.  It makes it useable in many more contexts. 

Also same note about static methods.